### PR TITLE
fix: now the app sidebar no longer crashes the page

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Fixes an infinite rerender issue in the application sidebar.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

This is only in beta at the moment, so I'm going to associate it to an existing issue.

Relates to #2477

## Steps to QA

1. Ensure you're logged in to the app
2. Navigate to https://deploy-preview-2506--oss-insights.netlify.app/workspaces/new
3. The page shouldn't crash anymore
4. Create a new workspace
5. The workspace is created and the new workspace's name appears in the workspaces dropdown in the app sidebar.
6. Update the name of the newly created workspace and click the `Update Workspace` button.
7. The workspace name gets updated in the dropdown in the app sidebar.

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: components/shared/AppSidebar/AppSidebar.tsx [hunk 1]
```diff
@@ -62,6 +59,24 @@ export const AppSideBar = () => {
     .slice(0, 5)
     .sort((a, b) => new Date(b.updated_at).getTime() - new Date(a.updated_at).getTime());
 
+  interface GlobalEventHandlersEventMap {
+    workspaceUpdated: CustomEvent;
+  }
+
+  useEffectOnce(() => {
+    const mutateHandler = (event: Event) => {
```
Comment (nickytonline) [inline components/shared/AppSidebar/AppSidebar.tsx:67]: Registers a custom event handler for updating workspaces in the drop down. This event gets dispatched when a new workspace is created or an existing one is updated. It calls useSWR's `mutate()` to update the workspaces list.

File: lib/store/index.ts [hunk 2]
```diff
@@ -11,7 +11,6 @@ const initialState: GlobalStateInterface = {
   userId: null,
   hasReports: false,
   openSearch: false,
-  workspaces: [],
```
Comment (nickytonline) [inline lib/store/index.ts:14]: We no longer need workspaces in Zustand.

File: components/shared/AppSidebar/AppSidebar.tsx [hunk 3]
```diff
@@ -41,18 +41,15 @@ const InsightsList = ({
   );
 };
 
+export const WORKSPACE_UPDATED_EVENT = "workspaceUpdated";
```
Comment (nickytonline) [inline components/shared/AppSidebar/AppSidebar.tsx:44]: I opted for the constant in this file as opposed to a seprate file. Maybe if we have more custom events in the future, it'd make sense to pull this out.

File: components/shared/AppSidebar/AppSidebar.tsx (hunk 1 above)
Comment (jpmcb) [reply components/shared/AppSidebar/AppSidebar.tsx:67]: Why the `// eslint-disable-next-line no-console` eslint skip here? I don't see a console log. Leftover from debugging?

**Review by jpmcb — APPROVED** [review]

File: components/shared/AppSidebar/AppSidebar.tsx (hunk 1 above)
Comment (nickytonline) [reply components/shared/AppSidebar/AppSidebar.tsx:67]: I opted for the custom event @brandonroberts becuase I could use `mutate()` from `useSWRConfig`, but I wanted to use the [bound mutate](https://swr.vercel.app/docs/mutation#bound-mutate) instead.

Comment (nickytonline) [reply components/shared/AppSidebar/AppSidebar.tsx:67]: It's here @jpmcb `event instanceof CustomEvent && console.info(WORKSPACE_UPDATED_EVENT, event.detail);`

Comment (nickytonline) [reply components/shared/AppSidebar/AppSidebar.tsx:67]: It's here @jpmcb, `event instanceof CustomEvent && console.info(WORKSPACE_UPDATED_EVENT, event.detail);`

File: components/shared/AppSidebar/AppSidebar.tsx [hunk 4]
```diff
@@ -62,6 +59,24 @@ export const AppSideBar = () => {
     .slice(0, 5)
     .sort((a, b) => new Date(b.updated_at).getTime() - new Date(a.updated_at).getTime());
 
+  interface GlobalEventHandlersEventMap {
+    workspaceUpdated: CustomEvent;
+  }
+
+  useEffectOnce(() => {
```
Comment (brandonroberts) [inline components/shared/AppSidebar/AppSidebar.tsx:66]: For my own knowledge, why `useEffectOnce` instead of `useEffect` with an empty array?

**Review by brandonroberts — APPROVED** [review]

File: components/shared/AppSidebar/AppSidebar.tsx (hunk 4 above)
Comment (nickytonline) [reply components/shared/AppSidebar/AppSidebar.tsx:66]: Pretty much. It's an alias.

File: components/shared/AppSidebar/AppSidebar.tsx (hunk 1 above)
Comment (jpmcb) [reply components/shared/AppSidebar/AppSidebar.tsx:67]: Ahhhhh missed the console log after the `&&` in that statment 🤦🏼 👍🏼

**Pull request merged**